VTKConverter/VTKConverter/DataImport/ModelData.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+using Kitware.VTK;
+
+namespace VTKConverter.DataImport
+{
+    abstract class ModelData
+    {
+        public double[] BoundingBox { get; protected set; }
+        public double[][] Vertices { get; protected set; }
+        public int NumberOfFacetEdges { get; protected set; }
+        public int[] Indices { get; protected set; }
+        public double[][] Vectors { get; protected set; }
+        public double[][] Scalars { get; protected set; }
+
+        public ModelData(vtkDataSet vtkModel)
+        {
+            GetBounds(vtkModel);
+        }
+
+        private void GetBounds(vtkDataSet vtkModel)
+        {
+            double[] boundingCoordinates = vtkModel.GetBounds();
+            BoundingBox = new double[6] {boundingCoordinates[0], boundingCoordinates[2], -boundingCoordinates[4],
+            boundingCoordinates[1], boundingCoordinates[3], -boundingCoordinates[5]};
+        }
+
+        protected void GetVertices(vtkDataSet vtkModel)
+        {
+            int numberOfPoints = vtkModel.GetNumberOfPoints();
+            Vertices = new double[numberOfPoints][];
+            for (int i = 0; i < numberOfPoints; i++)
+            {
+                Vertices[i] = vtkModel.GetPoint(i);
+                Vertices[i][2] = -Vertices[i][2];
+            }
+        }
+
+        protected virtual void GetIndices(vtkDataSet vtkModel)
+        {
+            int numberOfCells = vtkModel.GetNumberOfCells();
+            NumberOfFacetEdges = vtkModel.GetMaxCellSize();
+            Indices = new int[NumberOfFacetEdges * numberOfCells];
+            int currentIndexNumber = 0;
+            for (int i = 0; i < numberOfCells; i++)
+            {
+                currentIndexNumber = GetCellIndices(currentIndexNumber, vtkModel.GetCell(i).GetPointIds());
+            }
+        }
+
+        protected void SetPointIndices(int numberOfPoints)
+        {
+            NumberOfFacetEdges = 3;
+            Indices = new int[numberOfPoints * 3];
+            int currentVertexNumber = 0;
+            for (int i = 0; i < Indices.Length; i += 3)
+            {
+                for (int j = 0; j < 3; j++)
+                {
+                    Indices[i + j] = currentVertexNumber;
+                }
+                currentVertexNumber++;
+            }
+        }
+
+        private int GetCellIndices(int currentIndexNumber, vtkIdList cellIndices)

```

Comment:
Do you think it would be clearer/netter if the function returned Indices of a particular cell as a list and then the list would be added to the Indices array in a separate function?